packages/amplify-graphql-auth-transformer/src/resolvers/helpers.ts

```diff
@@ -252,6 +248,58 @@ export const generateOwnerClaimExpression = (ownerClaim: string, refName: string
   return compoundExpression(expressions);
 };
 
+/**
+ * Concatenates multiple owner claims if any
+ */
+export const generateOwnerMultiClaimExpression = (ownerClaim: string, refName: string): Expression => {
+  let expression: Expression;
+  const identityClaims = ownerClaim.split(IDENTITY_CLAIM_DELIMITER);
+  const hasMultiIdentityClaims = identityClaims.length > 1;
+
+  if (hasMultiIdentityClaims) {
+    const additionalClaims: string[] = [];
+    identityClaims.forEach((claim, idx) => {
```

Comment:
Rather than a foreach, this is potentially more undertstandable if we just write as a map. i.e.

```js
const additionalClaims = [...Array(identityClaims.length).keys()].splice(1).map((i) => `currentClaim${i}`);
return set(ref(refName), raw(`"$${[refName, ...additionalClaims].join(IDENTITY_CLAIM_DELIMITER)}"`));
```